### PR TITLE
[Events] Order and split current/past events

### PR DESCRIPTION
**Pull request by antonyharfield**
## ✅ DoD

- [x] All work is complete
- [x] Migrations: NA
- [x] Documentation: NA
- [x] Release notes: NA
- [x] Issues linked: resolves _none_

## 🛑 Problems

_None_

## 📝 Summary

- Add ordering to the events
- Split events page into current and past
- _Refactor event row to separate component_

## 💡 More ideas

_None_

## 📸 Screenshots

localhost_8080_events

### CODE REVIEW

**Comment by charles-allen** [comment]
Not your fault, but we might want to fix some:
- "Register now" doesn't work -- maybe hide it?
- Shouldn't show register for past events
- Event title & date feel too small; maybe needs some responsive love
- Tags look better in "Sentence Case"; perhaps we should fix (or just replace) the test-data

File: hosting/src/events/Events/EventRow.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Vue } from 'vue-class-component'
+import { Prop } from 'vue-property-decorator'
+
+import { EventUi } from '../models'
+
+export default class EventRow extends Vue {
+  @Prop()
+  event!: EventUi
```
Comment (charles-allen) [inline hosting/src/events/Events/EventRow.ts:8]: If you use the `vue-class-component` style, it has better TypeScript support:
```
export default class EventRow extends Vue.with(class {
  event: EventUi
}) {
```

File: hosting/src/events/Events/EventRow.html [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        ></div>
+      </div>
+      <p class="mt-3 text-center">
+        <router-link
+          v-if="true"
+          :to="{ name: 'Event', params: { slug: event.slug } }"
+          class="text-brand-blue-900 hover:text-brand-blue-500"
+          >&gt; &gt; Register Now &lt; &lt;</router-link
+        >
+      </p>
```
Comment (charles-allen) [inline hosting/src/events/Events/EventRow.html]: Broken "register now" link is the most important thing to hide right now

File: hosting/src/events/Events/Events.ts [hunk 3]
```diff
@@ -1,14 +1,24 @@
...
-    return this.$store.getters.events
+  get currentEvents(): EventUi[] {
+    const now = dayjs()
+    return this.$store.getters.events.filter((event: EventUi) => event.dateStart.isAfter(now)).reverse()
+  }
+
+  get pastEvents(): EventUi[] {
+    const now = dayjs()
+    return this.$store.getters.events.filter((event: EventUi) => event.dateStart.isBefore(now))
   }
```
Comment (charles-allen) [inline hosting/src/events/Events/Events.ts]: We might consider making these Vuex getters.
`Home` should be using `futureEvents` too.

**Review by charles-allen — APPROVED** [review]
Keep those changes coming!

lets-keep-moving

File: hosting/src/events/Events/EventRow.html (hunk 2 above)
Comment (antonyharfield) [reply hosting/src/events/Events/EventRow.html]: Done

File: hosting/src/events/Events/Events.ts (hunk 3 above)
Comment (antonyharfield) [reply hosting/src/events/Events/Events.ts]: Yep we should probably load them from Firebase independently too.

Comment (antonyharfield) [reply hosting/src/events/Events/Events.ts]: I did look at doing that but then I would have had to also change the way we get the event detail page using the slug.

**Pull request merged**